### PR TITLE
Adding a controller module to manage transitioning integrations to their desired status.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

**Comment by chirino** [comment]
I need to add some integration tests for this, but I think this implements the basic framework around managing the workflow state that @KurtStam, @iocanel and I discussed today.

File: model/src/main/java/com/redhat/ipaas/model/integration/Integration.java [hunk 1]
```diff
@@ -31,9 +31,7 @@
 @JsonDeserialize(builder = Integration.Builder.class)
 public interface Integration extends WithId<Integration>, WithName, Serializable {
 
-    public static enum Type {Activated, Deactivated, Draft};
-
-    public static enum Phase {Pending, Running, Succeeded, Failed, Unknown};
+    public static enum Status { Draft, Activated, Deactivated, Pending, Deleted};
```
Comment (KurtStam) [inline model/src/main/java/com/redhat/ipaas/model/integration/Integration.java]: I don't think you ever desire for the state to be pending. I'd prefer to keep them separated.

File: controllers/src/main/java/com/redhat/ipaas/controllers/IntegrationController.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    private void enqueueCheckIntegration(String id) {
+        executor.execute(() -> {
+            checkIntegration(dataManager.fetch(Integration.class, id));
+        });
+    }
+
+    private void checkIntegration(Integration integration) {
+        Integration.Status desired = integration.getDesiredStatus().orElse(Integration.Status.Draft);
+        Integration.Status current = integration.getCurrentStatus().orElse(Integration.Status.Draft);
+        if (current != desired) {
```
Comment (KurtStam) [inline controllers/src/main/java/com/redhat/ipaas/controllers/IntegrationController.java]: want to pull in/rebase and add the deployment data from https://github.com/redhat-ipaas/ipaas-rest/pull/231?

**Comment by KurtStam** [comment]
want to pull in/rebase and add the deployment data from https://github.com/redhat-ipaas/ipaas-rest/pull/231?

File: model/src/main/java/com/redhat/ipaas/model/integration/Integration.java (hunk 1 above)
Comment (chirino) [inline model/src/main/java/com/redhat/ipaas/model/integration/Integration.java]: Removing the Pending status.  current state should be set to null where Pending was being used before.

File: controllers/src/main/java/com/redhat/ipaas/controllers/WorkflowHandler.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Set;
+
+/**
+ *
+ */
+public interface WorkflowHandler {
+
+    Set<Integration.Status> getTriggerStatuses();
+
+    Integration.Status execute(Integration model) throws Exception;
```
Comment (rhipaasuser) [inline controllers/src/main/java/com/redhat/ipaas/controllers/WorkflowHandler.java]: I would ditch the `throws Exception` as we wrap all exception under IPaaSException.

**Pull request merged**